### ISSUE TITLE
Crash on loading saves and sometimes savestates

### ISSUE THREAD

**Issue by DarkD** [hrydgard/ppsspp#15970]
### Game or games this happens in

Any, but for example ULUS-10481 Yu-Gi-Oh! 5D's Tag Force 4 used

### What area of the game / PPSSPP

Anywhere game loads save, in case of Yu-Gi-Oh - right on boot.

### What should happen

Well, should load save, duh. That happens with old saves from before last working version and doesn't happen with new saves from version affected up to current one - v1.13.1-745-ga42807ea6

### Logs

[ppsspplog.txt](https://github.com/hrydgard/ppsspp/files/9489781/ppsspplog.txt)

### Platform

Windows

### Mobile phone model or graphics card

NVIDIA GeForce GTX 1650 SUPER

### PPSSPP version affected

v1.13.1-652-gcae8ce914

### Last working version

v1.13.1-648-gb5e5f780f

### Graphics backend (3D API)

Direct3D 11

### Checklist

- [X] Test in the [latest git build](https://buildbot.orphis.net/ppsspp/) in case it's already fixed.
- [X] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
- [X] Try resetting settings or older versions and include if the issue is related.
- [X] Try without any cheats and without loading any save states.
- [X] Include logs or screenshots of issue.

**Comment by hrydgard** [comment]
That is #15939 - maybe something with the paths? Hm

**Comment by hrydgard** [comment]
Can you upload a broken save? just zip and drag into the text window here on github.

**Comment by DarkD** [comment]
No, Texture Saving is off. I haven't used it for those games. I've grabbed VS installer and started it, but that'll take time...

**Comment by ANR2ME** [comment]
> No, Texture Saving is off. I haven't used it for those games. What components/workloads are needed for VS? It asks in installer. If none - I'll install it, if something big, few days later.

According to https://github.com/hrydgard/ppsspp/wiki/Build-instructions#prerequisites you will need the "Windows XP support for C++" but not sure whether this is still valid or not, XP support has been dropped as i remembered.

As i remembered, it need the "Desktop development with C++" and Windows SDK (if it's not already included)
![image](https://user-images.githubusercontent.com/7974720/188491977-3225ef42-bd32-4a2e-b872-66faa82e2a2d.png)

You will also need Python 3

**Comment by DarkD** [comment]
I've managed to compile debug build and when I'm loading save(I've put it in memstick folder of debug build) it throws this exception:
Exception thrown: write access violation.
**dest** was 0x11102B5D3DF0112.
Also attaching whole log from debug console. I hope it's not caused by admin rights or that was for naught, so I'll force admin rights on it and try again just in case.
[debuglog.txt](https://github.com/hrydgard/ppsspp/files/9497553/debuglog.txt)

**Comment by unknownbrackets** [comment]
When you crash, you should see something like this:

![Crash in Visual Studio IDE](https://user-images.githubusercontent.com/191233/188653761-296ab1fc-6ee1-480b-960b-0d7a34def7e9.png)

What I'm most interested in is that bottom right, "Call Stack" tab.  You can right click -> Select All, right click -> Copy.  It'll look like this:

```
>	PPSSPPWindows64.exe!TextureCacheCommon::Invalidate(unsigned int addr, int size, GPUInvalidationType type) Line 2180	C++
 	PPSSPPWindows64.exe!GPUCommon::InvalidateCache(unsigned int addr, int size, GPUInvalidationType type) Line 2998	C++
 	PPSSPPWindows64.exe!GPUCommon::PerformMemoryCopy(unsigned int dest, unsigned int src, int size) Line 2946	C++
 	PPSSPPWindows64.exe!FramebufferManagerCommon::DoSetRenderFrameBuffer(FramebufferHeuristicParams & params, unsigned int skipDrawReason) Line 504	C++
 	PPSSPPWindows64.exe!FramebufferManagerCommon::SetRenderFrameBuffer(bool framebufChanged, int skipDrawReason) Line 294	C++
 	PPSSPPWindows64.exe!GPUCommon::Execute_Prim(unsigned int op, unsigned int diff) Line 1702	C++
 	PPSSPPWindows64.exe!GPUCommon::FastRunLoop(DisplayList & list) Line 1078	C++
 	PPSSPPWindows64.exe!GPUCommon::InterpretList(DisplayList & list) Line 1027	C++
 	PPSSPPWindows64.exe!GPUCommon::ProcessDLQueue() Line 1234	C++
 	PPSSPPWindows64.exe!GPUCommon::UpdateStall(int listid, unsigned int newstall) Line 859	C++
 	[Inline Frame] PPSSPPWindows64.exe!sceGeListUpdateStallAddr(unsigned int displayListID, unsigned int stallAddress) Line 380	C++
 	PPSSPPWindows64.exe!WrapI_UU<&sceGeListUpdateStallAddr>() Line 201	C++
 	PPSSPPWindows64.exe!CallSyscallWithoutFlags(const HLEFunction * info) Line 659	C++
 	00007ff7d830f160()	Unknown
 	000001972deb0000()	Unknown
 	0000000009fbf980()	Unknown
 	PPSSPPWindows64.exe!00007ff7dc63d8f0()	C++
 	00000011dc9ff660()	Unknown
```

In this case, we can see where the crash came from.  The actual dest value of 0x11102B5D3DF0112 changes per system and run of PPSSPP, so doesn't tell me virtually anything.  Although, `dest` specifically is a bit interesting...

Also seeing what code it highlights (you can see it in my screenshot) would help, although not as much as the stack trace.

-[Unknown]

**Comment by DarkD** [comment]
Here you go:
>	PPSSPPDebug64.exe!DeIndexTexture4<unsigned int>(unsigned int * dest, const unsigned char * indexed, int length, const unsigned int * clut, unsigned int * outAlphaSum) Line 170	C++
 	PPSSPPDebug64.exe!TextureCacheCommon::DecodeTextureLevel(unsigned char * out, int outPitch, GETextureFormat format, GEPaletteFormat clutformat, unsigned int texaddr, int level, int bufw, bool reverseColors, bool expandTo32bit) Line 1526	C++
 	PPSSPPDebug64.exe!TextureCacheCommon::LoadTextureLevel(TexCacheEntry & entry, unsigned char * data, int stride, ReplacedTexture & replaced, int srcLevel, int scaleFactor, Draw::DataFormat dstFmt, bool reverseColors) Line 2489	C++
 	PPSSPPDebug64.exe!TextureCacheDX9::BuildTexture(TexCacheEntry * const entry) Line 303	C++
 	PPSSPPDebug64.exe!TextureCacheCommon::ApplyTexture() Line 1844	C++
 	PPSSPPDebug64.exe!DrawEngineDX9::DoFlush() Line 607	C++
 	PPSSPPDebug64.exe!DrawEngineDX9::Flush() Line 131	C++
 	PPSSPPDebug64.exe!DrawEngineDX9::DispatchFlush() Line 139	C++
 	PPSSPPDebug64.exe!GPUCommon::FastRunLoop(DisplayList & list) Line 1087	C++
 	PPSSPPDebug64.exe!GPUCommon::InterpretList(DisplayList & list) Line 1027	C++
 	PPSSPPDebug64.exe!GPUCommon::ProcessDLQueue() Line 1234	C++
 	PPSSPPDebug64.exe!GPUCommon::EnqueueList(unsigned int listpc, unsigned int stall, int subIntrBase, PSPPointer<PspGeListArgs> args, bool head) Line 822	C++
 	PPSSPPDebug64.exe!sceGeListEnQueue(unsigned int listAddress, unsigned int stallAddress, int callbackId, unsigned int optParamAddr) Line 342	C++
 	PPSSPPDebug64.exe!WrapU_UUIU<&sceGeListEnQueue>() Line 683	C++
 	PPSSPPDebug64.exe!CallSyscallWithoutFlags(const HLEFunction * info) Line 654	C++
 	[External Code]	
![image](https://user-images.githubusercontent.com/6061259/188656098-01e7a6da-e28a-4db2-a22f-43a9b0cfe793.png)

**Comment by unknownbrackets** [comment]
Okay, so that's not actually related to saves specifically but was probably triggered by drawing PPGe textures.  #15979 may fix it.  Also, does this only happen on Direct3D 9?

-[Unknown]

**Comment by DarkD** [comment]
Yeah, just checked, on debug build(which is by the way v1.13.1-818) it only happens on DX9. Happened everywhere on non-debug one. I'll check the other game I've mentioned, Gundam AGE, this was from running Yu-Gi-Oh.

**Comment by unknownbrackets** [comment]
If you update to the latest git, do you still get crashes?  Any different stack trace information?

Even if that wasn't the issue, thanks for catching it.  I was able to reproduce in another game in D3D9 (which tbh I do not use too often except when testing changes...)

-[Unknown]

**Comment by DarkD** [comment]
So today I wanted to confirm on non-debug 818 if it crashes same or not. I've used clean unpack without any changes to settings to be sure. Hadn't even placed save file in there. And 818 doesn't crash on DX9, OGL and Vulkan, but crashes on DX11 when loading animation disappears and caution message trying to appear. Same for Yu-Gi-Oh, boots with DX9, OGL, Vulkan, not DX11. (the one I couldn't confirm, of course)
Than I saw what you've wrote, [Unknown], and tried lastest build, it was 825 and it stopped crashing on it. Neither render crashes!
To be sure I've checked 820, build with your latest commit. And all works fine. Hooray!
So in the end even though I couldn't debug with DX11 your commit fixed it too. I have a feeling of some magic stuff going on there as you intended to fix DX9 and DX9 crashed on 818 debug, but bug disappeared and that's all I may wish for.
